ui/src/app/base/types.ts

```diff
@@ -1,8 +1,14 @@
 export type TSFixMe = any; // eslint-disable-line @typescript-eslint/no-explicit-any
 
-export type Sort = {
-  direction: "ascending" | "descending" | "none";
-  key: string;
+export enum SortDirection {
+  ASCENDING = "ascending",
+  DESCENDING = "descending",
+  NONE = "none",
+}
+
+export type Sort<K extends string | null = string> = {
+  direction: SortDirection;
```

Comment:
As above, using an object with `as const` could result in better TS suggestions.
```suggestion
  direction: typeof SortDirection[keyof typeof SortDirection]
```
before
```
(property) direction: SortDirection
```
after
```
(property) direction: "ascending" | "descending" | "none"
```